app/models/manageiq/providers/ibm_power_hmc/infra_manager.rb

```diff
@@ -20,6 +20,8 @@ class ManageIQ::Providers::IbmPowerHmc::InfraManager < ManageIQ::Providers::Infr
   supports :provisioning
 
   has_many :hosts_advanced_settings, :through => :hosts, :source => :advanced_settings
+  has_many :miq_scsi_targets, -> { distinct }, :through => :guest_devices
+  has_many :miq_scsi_luns, -> { distinct }, :through => :miq_scsi_targets
```

Comment:
@CallisteH I am moving these associations to core so that you don't have to define these here.  These can be removed once https://github.com/ManageIQ/manageiq/pull/21980 is merged.

Reply:
@agrare Not sure if this related, but I end up having new duplicate entries in the `miq_scsci_targets`/`miq_scsi_luns` tables every time I refresh the inventory. What do I have to do to ensure these entries remain unique ?

```
vmdb_development=# select * from miq_scsi_targets;
 id  | guest_device_id | target | iscsi_name | iscsi_alias | address |                  uid_ems
-----+-----------------+--------+------------+-------------+---------+--------------------------------------------
 530 |            1109 |        | vtscsi0    |             |         | 0862cc1c7985504a73
 531 |            1109 |        | vtopt0     |             |         | 195a07c107a921ac58
 532 |            1126 |        | vtscsi3    |             |         | 08d8e5257e750960dd
(...)
 544 |            1109 |        | vtscsi0    |             |         | 0862cc1c7985504a73
 545 |            1109 |        | vtopt0     |             |         | 195a07c107a921ac58
 546 |            1126 |        | vtscsi3    |             |         | 08d8e5257e750960dd
(...)
(28 rows)
````

Reply:
@vfebvre there are two things that typically cause this, one is the collection's association is misconfigured so it is returning `[]` for the existing records, the other is you aren't using the manager_ref correctly.  Since you have the guest_device_id and uid_ems the same I'm guessing it is the association.

I'm thinking the issue could be the parent_inventory_collections here: https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/inventory/persister/builder/infra_manager.rb#L213

Try deleting that and see if it fixes your issue.  If so I'll move that down to VMware

Reply:
Ohh right I keep forgetting these are for VMs not hosts